docs/3-v6-beta/4-flash-fill.md

```diff
@@ -1,9 +1,321 @@
+}
+
+pub fn flash_fill(ctx: Context<FlashFill>, amount: u64) -> Result<()> {
+    let instructions_sysvar = ctx.accounts.instructions_sysvar.to_account_info();
+
+    // check #1: make sure the current instruction isn't a cpi call
+    let current_idx = load_current_index_checked(&instructions_sysvar)? as usize;
+    let current_ixn = load_instruction_at_checked(current_idx, &instructions_sysvar)?;
+    require_keys_eq!(
+        current_ixn.program_id,
+        crate::ID
+    );
+
+    // check #2: check that the next instruction in the transaction performs a swap
+    let swap_ix_index = current_idx + 1;
+    if let Ok(jup_swap_ixn) = load_instruction_at_checked(swap_ix_index, &instructions_sysvar) {
+        // check #2a: check that it calls Jup v6 program
+        require_keys_eq!(
+            jup_swap_ixn.program_id,
+            Pubkey::from_str("JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4").unwrap()
+        );
+
+        // check #2b: check that it calls Jup's swap instruction
+        let ixn_identifier = u64::from_be_bytes(jup_swap_ixn.data[..8].try_into().unwrap());
+        require!(
+            ixn_identifier == get_instruction_discriminator(&[b"global:shared_accounts_route"])
+        );
+
+        // check #2c: check that the correct input and output mint is used for the swap
+        require_keys_eq!(
+            jup_swap_ixn.accounts[7].pubkey,
+            ctx.accounts.input_mint.key()
+        );
+        require_keys_eq!(
+            jup_swap_ixn.accounts[8].pubkey,
+            ctx.accounts.output_mint.key()
+        );
+
+        // check #2d: check that destinationTokenAccount for Jup ix
+        require_keys_eq!(
+            jup_swap_ixn.accounts[6].pubkey,
+            ctx.accounts.escrow_output_token_account.key()
+        );
+
+        // check #2e: check that the full amount of input_mint is used to swap
+        let swap_amount_start_at = jup_swap_ixn.data.len() - 19;
+        let swap_amount = u64::from_le_bytes(
+            jup_swap_ixn.data[swap_amount_start_at..swap_amount_start_at + 8]
+                .try_into()
+                .unwrap(),
+        );
+        require_eq!(swap_amount, actual_swap_amount);
+    } else {
+        return Err(error!(MissingSwapInstructions)); // create your own error here
+    }
+
+    require_gte!(ctx.accounts.escrow_input_token_account.amount, amount);
+
+    let signer_seeds: &[&[&[u8]]] = &[...];
+    let now = Clock::get()?.unix_timestamp;
+
+    // transfer to admin for swap
+    anchor_spl::token::transfer(
+        CpiContext::new_with_signer(
+            ctx.accounts.token_program.to_account_info(),
+            anchor_spl::token::Transfer {
+                from: ctx.accounts.escrow_input_token_account.to_account_info(),
+                to: ctx.accounts.admin_input_token_account.to_account_info(),
+                authority: ctx.accounts.escrow.to_account_info(),
+            },
+            signer_seeds,
+        ),
+        amount,
+    )?;
+
+    Ok(())
+}
```

Comment:
we should add the instruction code after flashfill? that is very important I think.